SingularityUI/app/components/machines/Racks.jsx

```diff
@@ -17,49 +17,29 @@ const typeName = {
   'decommissioning': 'Decommissioned By'
 };
 
-class Racks extends Component {
-
-  static propTypes = {
-    racks: PropTypes.arrayOf(PropTypes.shape({
-      state: PropTypes.string
-    })),
-    decommissionRack: PropTypes.func.isRequired,
-    removeRack: PropTypes.func.isRequired,
-    reactivateRack: PropTypes.func.isRequired,
-    clear: PropTypes.func.isRequired,
-    error: PropTypes.string
-  };
-
-  componentWillUnmount() {
-    this.props.clear();
-  }
-
-  showUser(rack) {
-    return Utils.isIn(rack.currentState.state, ['ACTIVE', 'DECOMMISSIONING', 'DECOMMISSIONED', 'STARTING_DECOMMISSION']);
-  }
-
-  getMaybeReactivateButton(rack) {
-    return (Utils.isIn(rack.currentState.state, ['DECOMMISSIONING', 'DECOMMISSIONED', 'STARTING_DECOMMISSION']) &&
-      <ModalButton
-        buttonChildren={<Glyphicon glyph="new-window" />}
-        action="Reactivate Rack"
-        onConfirm={(data) => this.props.reactivateRack(rack, data.message)}
-        tooltipText={`Reactivate ${rack.id}`}
-        formElements={[messageElement]}>
-        <p>Are you sure you want to cancel decommission and reactivate this rack??</p>
-        <pre>{rack.id}</pre>
-        <p>Reactivating a rack will cancel the decommission without erasing the rack's history and move it back to the active state.</p>
-      </ModalButton>
-    );
-  }
-
-  getDecommissionOrRemoveButton(rack) {
+const Racks = (props) => {
+  const showUser = (rack) => Utils.isIn(rack.currentState.state, ['ACTIVE', 'DECOMMISSIONING', 'DECOMMISSIONED', 'STARTING_DECOMMISSION']);
+
+  const getMaybeReactivateButton = (rack) => (Utils.isIn(rack.currentState.state, ['DECOMMISSIONING', 'DECOMMISSIONED', 'STARTING_DECOMMISSION']) &&
```

Comment:
Might want to split these into their own functions, its difficult to follow what's going on here.